### ISSUE TITLE
Feature request: Categories for Aliases.

### ISSUE THREAD

**Issue by TheSagest** [opnsense/core#3083]
I would like to see "categories" in the Aliases GUI as there is for firewall Rules.

**Comment by fichtner** [comment]
Would you mind describing the desired functionality? There is not enough information here to design or implement something and that way it’s hard to assess required work, impact and usefulness. Thanks.

**Comment by TheSagest** [comment]
My use case for this is as follows.

I am using the URL table for IPs for my clients OPNsense boxes. I will manage just 1 table for IP’s (say, similar to bogon IPs) I will also have a individual client URL tables which I update at one site. So everything is one place.

I would like to see a Port(s) Table to be the same, so I can update a single file and it will be downloaded and updated on ALL client firewalls.

Example.
Client A has a PBX system
I have several PBX URL Table(s) for known External PBX IP addresses,

If they change PBX systems, I just change the URL in the Alias definition on their OPNSense.
( I know there Is only 1 IP, but for a different scenario there may lots)

So to recap, I will manage ONE file with required ports and share it with many clients using a URL Table(ports)

Thanks
The Sage !!

P.S.
It is sort of like DNS where it is easier to understand and remember English Names than IP addresses. Other authorised users can login and will be able to see what is happening easier.

**Comment by TheSagest** [comment]
This is my original plan, which led me to the feature request, as I was going to use the API to


  1.  FLUSH the table
  2.  Then re add the updated IP’s, Ports

2 things happened,

  1.  FLUSH API did not seem to work, It also didn’t work by going to Diag., pfTables. ->flush.
  2.  If FLUSH did work, adding a IP address to an empty Alias work error out as (can’t find post right now)  someelse mentioned. It does work if there is already one in there.

That’s when I had the great idea to use URL tables (IP) and found Ports wasn’t an option yet.

Which led to the Feature request. I assume, as the FIREWALL API is still in development, a feature request will be premature.

The Sage !!

**Comment by AdSchellevis** [comment]
This issue has been automatically timed-out (after 180 days of inactivity).

For more information about the policies for this repository, 
please read https://github.com/opnsense/core/blob/master/CONTRIBUTING.md for further details.

If someone wants to step up and work on this issue, 
just let us know, so we can reopen the issue and assign an owner to it.
